### ISSUE TITLE
Name property in EventBridgeRule doesn't exist in schema.json

### ISSUE THREAD

**Issue by francisoud** [aws/serverless-application-model#3086]
### Description
Name property in EventBridgeRule doesn't exist in schema.json

From the documentation: https://docs.aws.amazon.com/serverless-application-model/latest/developerguide/sam-property-statemachine-statemachineeventbridgerule.html#sam-statemachine-statemachineeventbridgerule-name

> Name
> The name of the rule.
> Type: String
> Required: No
> AWS CloudFormation compatibility: This property is passed directly to the [Name](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-events-rule.html#cfn-events-rule-name) property of an AWS::Events::Rule resource.

But no such thing in the schema.json
https://github.com/aws/serverless-application-model/blob/develop/samtranslator/validator/sam_schema/schema.json#L362

### Steps to reproduce
generate sample project

```
~/$ sam init --name sam-eventbridge --runtime python3.9 
Which template source would you like to use?
        1 - AWS Quick Start Templates
        2 - Custom Template Location
Choice: 1

Choose an AWS Quick Start application template
        1 - Hello World Example
        2 - Hello World Example With Powertools
        3 - Infrastructure event management
        4 - Multi-step workflow
        5 - Lambda EFS example
        6 - Serverless Connector Hello World Example
        7 - Multi-step workflow with Connectors
Template: 7

Based on your selections, the only Package type available is Zip.
We will proceed to selecting the Package type as Zip.

Based on your selections, the only dependency manager available is pip.
We will proceed copying the template using pip.

Would you like to enable X-Ray tracing on the function(s) in your application?  [y/N]: N

Would you like to enable monitoring using CloudWatch Application Insights?
For more info, please view https://docs.aws.amazon.com/AmazonCloudWatch/latest/monitoring/cloudwatch-application-insights.html [y/N]: N

    -----------------------
    Generating application:
    -----------------------
    Name: sam-eventbridge
    Runtime: python3.9
    Architectures: x86_64
    Dependency Manager: pip
    Application Template: step-functions-with-connectors
    Output Directory: .
    Configuration file: sam-eventbridge/samconfig.toml
    
    Next steps can be found in the README file at sam-eventbridge/README.md
```

Add:
```
Resources:
  EventBus:
    Type: AWS::Events::EventBus
    Properties:
      Name: "my-eventbus"
```

Replace
```
StockTradingStateMachine:
    Type: AWS::Serverless::StateMachine 
    DependsOn: EventBus
    Properties:
      DefinitionUri: statemachine/stock_trader.asl.json
      DefinitionSubstitutions:
        StockCheckerFunctionArn: !GetAtt StockCheckerFunction.Arn
        StockSellerFunctionArn: !GetAtt StockSellerFunction.Arn
        StockBuyerFunctionArn: !GetAtt StockBuyerFunction.Arn
        DDBPutItem: !Sub arn:${AWS::Partition}:states:::dynamodb:putItem
        DDBTable: !Ref TransactionTable
      Events:
        EventRule:
          Type: EventBridgeRule
          Properties:
            Name: "my-eventrule-name"
            EventBusName: "my-eventbus"
            Pattern:
              detail:
                UserID:
                  - "XXXXXX"
            State: ENABLED
```

### Observed result
```
~/sam-eventbridge$ sam validate --lint
E0001 Error transforming template: Resource with id [StockTradingStateMachineEventRule] is invalid. property Name not defined for resource of type EventBridgeRule
/home/benjamin/test-sam-python/sam-eventbridge/template.yaml:1:1
```

### Expected result
Same behavior as descibe in documentation

### Additional environment details
```
$ sam --info
{
  "version": "1.78.0",
  "system": {
    "python": "3.7.10",
    "os": "Linux-5.15.90.1-microsoft-standard-WSL2-x86_64-with-debian-bookworm-sid"
  },
  "additional_dependencies": {
    "docker_engine": "20.10.21",
    "aws_cdk": "Not available",
    "terraform": "Not available"
  }
}
```

**Comment by aaythapa** [comment]
Hi there, thank you for reporting the issue. I can confirm that you're correct, we seemed to have missed the `Name` property in `EventsBridgeRule` event in SAM Statemachine. Will add that to the schema. Thanks again.

**Comment by aaythapa** [comment]
The PR has been merged so closing this issue, please let us know if the issue persists 👍

**Comment by davidjb** [comment]
@aaythapa The issue still persists despite the PR merged above - I've logged an issue at #3215 and described where and why the problem is occurring.
